[DOCS] Add dev mode next step (#20102)
ethereum/go-ethereum: docs/_getting-started/dev-mode.md

---
title: Dev mode
---

Geth has a development mode which sets up a single node Ethereum test network with a number of options optimized for developing on local machines. You enable it with the `--dev` argument.

Starting geth in dev mode does the following:

-   Initializes the data directory with a testing genesis block
-   Sets max peers to 0
-   Turns off discovery by other nodes
-   Sets the gas price to 0

> -   Uses the ethash consensus engine with clique headers to allows blocks to be mined as fast as your CPU and memory can handle

Reviewer: It's a separate consensus engine, 
```suggestion
-   Uses the Clique PoA consensus engine with which allows blocks to be mined as-needed without excessive CPU and memory consumption
```

> -   Prevents the consensus (proof of work) difficulty from ever increasing

Reviewer: Nah, strike this. It doesn't use PoW, but there is a notion of `difficulty` which is used by Clique to figure out which chain is canonical, but that's a bit too much detail probably

Author: ```suggestion
```

> -   Permits 0 block times in the clique consensus engine and forbid empty blocks for 0 periods

Reviewer: 0 block times - not sure what that means. It's possible to set `dev.period` to have e.g. one block every second, but by default it mines if there are any transactions waiting to be included.

Author: I can't quite remember where this information came from, and I don't even understand it myself anymore now.

I think it was something to do with speeding up testing by effectively disabling features of the consensys engine. Is it best to cut this line or paraphrase what you say above?

Reviewer: Write "Uses on-demand block generation, producing blocks when there are transactions waiting to be mined"

You can specify a data directory to maintain state between runs using the `--datadir` option, otherwise databases are ephemeral and in-memory:

```shell
$ mkdir test-chain-dir
$ geth --dev --datadir test-chain-dir console
```

Once geth is running in dev mode, you can interact with it in the same way as when geth is running in other ways.

For example, create a test account:

```shell
> personal.newAccount()
```

Then transfer ether from the coinbase to the new account:

```shell
> eth.sendTransaction({from:eth.coinbase, to:eth.accounts[1], value: web3.toWei(0.05, "ether")})
```

And check the balance of the account:

```shell
> eth.getBalance(eth.accounts[1])
```

If you want to test your dapps with a realistic block time use the `--dev.period` option when you start dev mode:

```shell

> geth --dev --dev.period 100 console

Reviewer: ```suggestion
geth --dev --dev.period 14 console
```
14 is more realistic than 100
